### ISSUE TITLE
Classic speculative verification differs from llama.cpp for explicit top_p: 0

### ISSUE THREAD

**Issue by ardan-bkennedy** [ardanlabs/kronk#936]
## Summary

Classic speculative verification reconstructs the target sampling distribution in Go. For an explicitly requested `top_p: 0`, that reconstruction appears to differ from the native llama.cpp sampler chain.

This should be resolved from differential evidence against llama.cpp rather than by assuming which behavior is intended.

Related performance investigation: #934.

## Reachable input

Kronk deliberately preserves an explicit zero:

- `TestParseParamsTopP` covers `top_p: 0` overriding the model default.
- `adjustParams` replaces zero only when `top_p` was not explicitly requested.
- Both `toSampler` and `buildDraftSampler` pass the preserved value to `llama.SamplerInitTopP(p.TopP, 0)`.

## Apparent mismatch

In llama.cpp b10819, the native top-p sampler treats only `p >= 1` as a no-op. For `p == 0` and `min_keep == 0`, it sorts candidates and stops after including the first candidate because the cumulative probability is already `>= 0`. The resulting candidate set therefore contains one token.

Kronk's classic speculative target reconstruction in `applySamplerFilters` applies top-p only when:

```go
if topP > 0 && topP < 1.0 {
```

For `topP == 0`, it skips top-p and retains the distribution left by top-k/min-p. That distribution is used as `p_target` by probabilistic classic verification, while the draft and ordinary target sampler chains use llama.cpp's native behavior.

Greedy requests and non-speculative generation do not use this Go reconstruction.

## Expected work

Add differential tests that apply identical deterministic logits and sampling parameters to:

1. llama.cpp's native `top-k → top-p → min-p → temperature` sampler chain; and
2. Kronk's `applySamplerFilters` reconstruction.

The tests should compare candidate support and normalized probabilities, including at least:

- `top_p`: `0`, a small positive value, the normal range, and `1`;
- `top_k`: a positive value, `0`, and `-1`;
- interactions with min-p and suppressed tokens.

Once the native behavior is pinned by those tests, align the Go reconstruction without changing the public meaning of sampling parameters speculatively.

**Comment by ardan-bkennedy** [comment]
Fixed with PR https://github.com/ardanlabs/kronk/pull/945
